presto-memory/src/test/java/io/prestosql/plugin/memory/TestMemorySmoke.java

```diff
@@ -95,40 +95,55 @@ public void testSelect()
     }
 
     @Test
-    public void testJoinDynamicFilteringNone()
```

Comment:
please also add test cases to `AbstractCoordinatorDynamicFilteringTest`

Reply:
Thanks for the pointer; added. Also made some changes to `DynamicFilterService` to register semi-join dynamic filters. Rebased to master to fix a file conflict.